### ISSUE TITLE
Delete source input assemblies

### ISSUE THREAD

**Issue by Albeoris** [emerbrito/ILMerge-MSBuild-Task#15]
**Workflow:**
I'm writing a small installer that needs to be packed into one executable file to not to scare users with the number of assemblies. I don't need original assemblies after the merge.

**Desire:**
It would be great to be able to automatically delete them. Now I redirected the output folder to another location, and set the OutputFile property, but this does not allow using the Output directory provided by the project settings.

P.S. Thanks for the great solution!

**Comment by emerbrito** [comment]
Do you need the project output directory to have only the merged assembly and nothing else? I believe we can add a parameter in the configuration file to control that.  Something like this:

- Merge assemblies with a temporary file name (to avoid conflicting with the project output)
- Clear directory (except for the file created by the task)
- Rename file

Let me know if this fits what you have in mind.

**Comment by Albeoris** [comment]
This is too dangerous a solution. Especially when you consider that the output directory can be anything. For example, a system disk. No, only those files that have been merged must be deleted.
It seems that this should not be difficult because we pass them to the ILMerge.

**Comment by emerbrito** [comment]
I agree. But considering you may be left with additional files in the output directory, not only your merged file, it is really helping you in any way besides clearing a bit of clutter? You will eventually have to grab what you want out of this output anyway (you usually don't have a reason to keep this folder contents for long) so at the end, would this bring any actual value?

**Comment by Albeoris** [comment]
If there are files in the output directory that should be in due to the copy settings, we have no reason to delete them.
If we want to merge several assemblies into one, then most likely we don't need non-merged assemblies in the output folder.
In my case, the output folder doesn't contain anything other than the files that should be present. It can be copied without additional cleaning. :)

**Comment by emerbrito** [comment]
After giving it some thought I decided not to implement this for a couple of reasons, including but not limited to:

- Attempting to delete the content of the output folder may be problematic if you have additional build tasks that may rely on them.
- If will fail if least one of those files is locked.

By default the merged assembly is already added to a sub-folder under under your $(TargetDir).
Here is an example:

![image](https://user-images.githubusercontent.com/5498239/51199519-f5529f00-18c4-11e9-87f8-aa2bfe4669ce.png)

The content of the "Debug" folder is untouched.
The merged assembled is under the ILMerge folder:

![image](https://user-images.githubusercontent.com/5498239/51199565-14e9c780-18c5-11e9-84b9-a2fe7d3c7c6e.png)

This folder is clean since it contains only the merged output.
It can still be accessed using build properties/variables such as: $(TargetDir)\ILMerge.

I'm closing this for now.
If you still believe this is a good addition to the library feel free to re-open with additional details.

**Comment by ckovamees** [comment]
I'm ~~reopening~~ (seems like I couldn't re-open, so see issue #28) and submitting a pull-request.

Thanks for a great project, I found it searching for a replacement to the ILMerge MSBuild NuGet package by Alexander Nosenko, as it seems it's not maintained anymore.

I've implemented builds using the default to that NuGet Package - which is to delete the CopyTo Assemblies and overwrite the Target. This leaves a nice TargetDirectory (bin\Debug or bin\Release). I know the ILMerge sub-directory contains the OutputFile, but thought that this change was easier on my builds.

So I added the advanced property "DeleteCopiesOverwriteTarget". If set to true the pull request will delete any CopyTo assemblies in the TargetDir and then overwrite the TargetFile with the merged one from OutputFile.
